.binder/environment.yml

```diff
@@ -2,42 +2,42 @@ channels:
 - conda-forge
 - nodefaults
 dependencies:
-# !! you can only use single `=` to set a version. Otherwise it will break the update job
-- invoke=2.2.0
-- packaging
-- pyyaml
+- ruamel.yaml
 # applications
-- jupyterlab=4.1.8
-- jupyter-collaboration=2.1.1
-- nbconvert=7.16.4
-- notebook=7.1.3
+- jupyterlab
```

Comment:
Do we want to pin it to `>=4.3.5`? Currently it resolves to 

![image](https://github.com/user-attachments/assets/989129d2-c641-485b-981c-97e5ad3b4d7d)

If we are giving a demo of jupyterlab, it feels like this one package should probably be pinned as we would not want to receive reports for things already fixed in newer versions?

Reply:
@krassowski I dug into why it was resolving to this, and it's because that's the base version in repo2docker - last updated in https://github.com/jupyterhub/repo2docker/pull/1369. I agree that we should try to use latest jupyterlab here if possible. If we pin to >=4.3.5, it means that when repo2docker goes to a version above 4.3.5, it'll simply just get pinned to that rather than latest. Do you know if there's a way to specify 'latest' with a conda environment.yml?

Reply:
@krassowski ok, so I've made https://github.com/jupyterhub/repo2docker/pull/1407 which bumps up all versions of things, including lab to 4.3.5. For this PR, my suggestion is that we either find a way to specify 'latest' or let it be unpinned, as once that repo2docker PR lands >=4.3.5 is a no-op

Reply:
I've checked with a colleague working on conda and it seems there is no way to specify `latest` but if we had an option to customize the CLI invocation we could use `--prune` to ignore constraints from currently installed versions.